src/adapters/uniswap-v3/products/pool/uniswapV3PoolAdapter.ts

```diff
@@ -0,0 +1,354 @@
+import { ethers, formatUnits } from 'ethers'
+import { Chain } from '../../../../core/constants/chains'
+import {
+  IMetadataBuilder,
+  CacheToFile,
+} from '../../../../core/decorators/cacheToFile'
+import { filterMap } from '../../../../core/utils/filters'
+import { getTokenMetadata } from '../../../../core/utils/getTokenMetadata'
+import {
+  ProtocolAdapterParams,
+  ProtocolDetails,
+  PositionType,
+  GetPositionsInput,
+  GetEventsInput,
+  MovementsByBlock,
+  GetTotalValueLockedInput,
+  GetProfitsInput,
+  GetApyInput,
+  GetAprInput,
+  GetClaimableRewardsInput,
+  GetConversionRateInput,
+  ProtocolRewardPosition,
+  ProtocolTokenApr,
+  ProtocolTokenApy,
+  ProtocolTokenUnderlyingRate,
+  ProfitsWithRange,
+  ProtocolTokenTvl,
+  ProtocolPosition,
+  TokenType,
+  Underlying,
+} from '../../../../types/adapter'
+import { Erc20Metadata } from '../../../../types/erc20Metadata'
+import { IProtocolAdapter } from '../../../../types/IProtocolAdapter'
+import { Protocol } from '../../../protocols'
+import { PositionManager__factory } from '../../contracts'
+
+const deadline = Math.floor(Date.now() - 1000) + 60 * 10
```

Comment:
A comment with what this calculation means would be handy.

Reply:
added